From abd33157600189a70a9d9438757c7f7831dc31ca Mon Sep 17 00:00:00 2001
From: Christian Svensson <bluecmd@google.com>
Date: Sun, 31 Mar 2019 14:13:45 -0700
Subject: [PATCH 3/4] tmp421: Register sensor with thermal system

hwmon expects the first info channel to be a chip configuration for it
to consider it for registration with the thermal system. Add that
channel configuration and set the thermal registration bit.

Signed-off-by: Christian Svensson <bluecmd@google.com>
---
 drivers/hwmon/tmp421.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/hwmon/tmp421.c b/drivers/hwmon/tmp421.c
index a94e35cff3e5..e7ad9a03b265 100644
--- a/drivers/hwmon/tmp421.c
+++ b/drivers/hwmon/tmp421.c
@@ -89,9 +89,11 @@ MODULE_DEVICE_TABLE(of, tmp421_of_match);
 struct tmp421_data {
 	struct i2c_client *client;
 	struct mutex update_lock;
+	u32 chip_config[2];
 	u32 temp_config[5];
+	struct hwmon_channel_info chip_info;
 	struct hwmon_channel_info temp_info;
-	const struct hwmon_channel_info *info[2];
+	const struct hwmon_channel_info *info[3];
 	struct hwmon_chip_info chip;
 	char valid;
 	unsigned long last_updated;
@@ -302,13 +304,19 @@ static int tmp421_probe(struct i2c_client *client,
 	if (err)
 		return err;
 
+	data->chip_config[0] = HWMON_C_REGISTER_TZ;
+
 	for (i = 0; i < data->channels; i++)
 		data->temp_config[i] = HWMON_T_INPUT | HWMON_T_FAULT;
 
 	data->chip.ops = &tmp421_ops;
 	data->chip.info = data->info;
 
-	data->info[0] = &data->temp_info;
+	data->info[0] = &data->chip_info;
+	data->info[1] = &data->temp_info;
+
+	data->chip_info.type = hwmon_chip;
+	data->chip_info.config = data->chip_config;
 
 	data->temp_info.type = hwmon_temp;
 	data->temp_info.config = data->temp_config;
-- 
2.23.0.187.g17f5b7556c-goog

